tests/test-pwm.js

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) 2016, Intel Corporation.
+
+
+console.log("Wire IO3 to IO2");
+
+var total = 0;
+var passed = 0;
+
+function assert(actual, description) {
+    total += 1;
+
+    var label = "\033[1m\033[31mFAIL\033[0m";
+    if (actual === true) {
+        passed += 1;
+        label = "\033[1m\033[32mPASS\033[0m";
+    }
+
+    console.log(label + " - " + description);
+}
+
+function expectThrow(description, func) {
+    var threw = false;
+    try {
+        func();
+    }
+    catch (err) {
+        threw = true;
+    }
+    assert(threw, description);
+}
+
+var pwm = require("pwm");
+var gpio = require("gpio");
+var pins = require("arduino101_pins");
+var pinA, pinB, msTimer, cycleTimer;
+
+pinB = gpio.open({pin: pins.IO2, direction: 'in'});
+
+// PWMPins open
+pinA = pwm.open({channel: pins.IO3});
+assert(pinA != null && typeof pinA == "object",
+      "open: defined pin and default argument");
+
+expectThrow("open: undefined pin", function () {
+    pinA = pwm.open({channel: 1024});
+});
+
+// set Period and PulseWidth with ms
+// duty cycle: 30%
+var msTrue = 0;
+var msFalse = 0;
+var msCount = 0;
+
+expectThrow("pwmpin: set pulseWidth without period", function () {
+    pinA.setPulseWidth(300);
```

Comment:
This does not throw, again because there's not requirement that you set period first and then pulse width first. I guess there could be. So I'd remove this test. The way we're going to fix it is that I've asked Zephyr to put both values into the same call, as in set_pwm_cycles(period, pulse_width). That way you're never in an invalid, halfway defined state. We'll change our API to match and then we'll be able to throw on errors.